src/cnlpt/cnlp_processors.py

```diff
@@ -75,91 +157,78 @@ def __init__(self, data_dir:str, tasks:Set[str]=None):
             ## user specified at the cli, remove those tasks so we don't also get them from other datasets
             ## and overwrite these.
             first_split = next(iter(self.dataset.values()))
-            dataset_tasks = tasks.intersection(first_split.features.keys())
-            tasks -= dataset_tasks
-            dataset_tasks = list(dataset_tasks)
-            dataset_tasks.sort()
-            metadata = {'tasks': dataset_tasks}
+            dataset_tasks = first_split.features.keys() - set(['text', 'text_a', 'text_b'])
+            active_tasks = tasks.intersection(dataset_tasks)
+            tasks -= active_tasks
```

Comment:
the longer term idea was something like, the arguments to the --task flag could be spread across multiple datasets, and for each dataset that you process with an autoprocessor, it removes the tasks from that array, essentially "claiming" them. Not sure if this is a bad way to do it. And it also ignores the setting where 2 datasets have the same task name (where you might want it to be trained in both if the labels are the same). Suggestions? Maybe for now I should just leave the parameter value the same and not try to have datasets claim tasks.

Reply:
ah, i see. i like the idea but i think it is complicated by the point about colliding task names. i *think* you can get around that by preserving the set of tasks and just taking the intersection each time, since the result should be the same except in the case of colliding names, right?

on the point of colliding names, though, if we do address that, are we thinking the command line argument for --task would look like `--task task1 task1 task2`, where `task1` is in two datasets, or just `--task task1 task2`, and the processors for both datasets would pick up `task1`?

also, what is the goal of having processors claim tasks?

Reply:
for the colliding names, i was thinking the latter. the point of claiming tasks is so that you can warn if they supply a task that you couldn't find in the data

Reply:
maybe you could do it by returning the tasks you found in the dataset and let the caller check whether their task was found